userComponent/src/main/kotlin/com/gravatar/app/usercomponent/data/WordPressClient.kt

```diff
@@ -0,0 +1,63 @@
+package com.gravatar.app.usercomponent.data
+
+import com.gravatar.app.foundations.DispatcherProvider
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.forms.submitForm
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.http.parameters
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+internal class WordPressClient(
+    private val httpClient: HttpClient,
+    private val dispatcherProvider: DispatcherProvider,
+) {
+
+    companion object {
+        private const val BASE_URL = "https://public-api.wordpress.com"
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    suspend fun login(
+        code: String,
+        clientSecret: String,
+        redirectUri: String,
+        clientId: String
+    ): Result<String> = withContext(dispatcherProvider.io) {
```

Comment:
Standard `Result` for now - but I want to introduce our own type for better error types, rather than exceptions.